x-pack/plugins/apm/server/routes/services/get_services_detailed_statistics/get_service_aggregated_transaction_detailed_statistics.ts

```diff
@@ -141,13 +141,17 @@ export async function getServiceAggregatedTransactionDetailedStats({
   );
 
   return keyBy(
-    response.aggregations?.sample.services.buckets.map((bucket) => {
+    response.aggregations?.sample.services.buckets.forEach((bucket) => {
```

Comment:
are you sure you want to use `forEach` here?

Reply:
Hm,   don't see any obvious reason not to use but now I'm thinking that `forEach` mutates the array. Let me double check this